src/components/tutorial/ArraysTutorial.tsx

```diff
@@ -23,6 +36,47 @@ const ArraysTutorial: React.FC<ArraysTutorialProps> = (props) => {
     console.log(example.propertyOne);
     console.log(example.methodOne());
 
+export const ArraysTutorial: React.FC = () => {
+  const [list, setList] = useState<TodoItem[]>([{id: 0, value: ''}])
+
+  const handleChange = (value: string, id: TodoItem['id']) => {
+    setList(prev => prev.map(item => item.id === id ? {... item, value} : item))
+  }
+
+  const handleDelete = (id: TodoItem['id']) => {
+  setList(prev => prev.filter(item => item.id !== id))
+  }
+
+  const handleAdd = (index; number) => {
+    const newItem = {id: count++, value: ''}
+    setList(prev => [...prev.slice(0, index + 1), newItem, ...prev.slice(index + 1)])
+  }
+
+  return (
+    <div>
+    {list.map((item, index) => (
+      <div key={item.id}>
+      <TextField
+        value={item.value}
+        onChange={e => handleChange(e.currentTarget.value, item.id)}
+      />
+      <IconButton onclick={() => handleAdd(index)}>
+         <AddIcon />
+      </IconButton>
+      {list.length > 1 && (
```

Comment:
I don't understand this IF statement, why is it necessary? Why do you need to know the size of the list while checking a single item?